android/engine/src/main/java/org/smartregister/fhircore/engine/util/ReflectionUtil.kt

```diff
@@ -46,3 +48,22 @@ suspend fun <T : Any> KClass<T>.callSuspendFunctionOnField(
       .apply { isAccessible = true }
   return declaredObjectMethod.callSuspend(declaredObject, *args)
 }
+
+inline fun <reified T : Any> getPrivateProperty(property: String, obj: T): Any? {
```

Comment:
Was this implemented as workaround? Reflection code is less performant and would require code refactoring on API changes.